odoo/tools/pdf.py

```diff
@@ -23,6 +22,34 @@ def _unwrapping_get(self, key, default=None):
 DictionaryObject.get = _unwrapping_get
 
 
+def readDigitUntilWhitespace(stream, maxchars=None):
+    """
+    Reads non-whitespace characters and returns them.
+    Stops upon encountering whitespace or when maxchars is reached.
+
+    Monkey patch readUntilWhitespace: only read digits.
+    https://github.com/mstamy2/PyPDF2/issues/164
+    https://github.com/mstamy2/PyPDF2/issues/183
+    """
+    txt = b""
+    while True:
+        tok = stream.read(1)
+        if txt and tok.isspace() or not tok:
+            break
+        # must be int
+        try:
+            int(tok)
```

Comment:
`tok.isdigit()`  is probably cheaper, and certainly clearer. Or `tok in string.digits` if we specifically only want to include western arabic numerals (0-10).

Reply:
`tok.isdecimal()` then because what we want to do is form numbers in base 10
https://docs.python.org/3/library/stdtypes.html#str.isdecimal